der/derive/src/enumerated.rs

```diff
@@ -36,13 +39,25 @@ impl DeriveEnumerated {
 
         // Reject `asn1` attributes, parse the `repr` attribute
         let mut repr: Option<Ident> = None;
+        let mut integer = false;
 
         for attr in &input.attrs {
             if attr.path.is_ident(ATTR_NAME) {
-                abort!(
-                    attr.path,
-                    "`asn1` attribute is not allowed on `Enumerated` types"
-                );
+                if let Ok(Meta::List(MetaList { nested, .. })) = attr.parse_meta() {
+                    for meta in nested {
+                        if let NestedMeta::Meta(Meta::NameValue(nv)) = meta {
+                            if nv.path.is_ident("type") {
+                                if let Lit::Str(lit) = nv.lit {
+                                    match lit.value().as_str() {
+                                        "ENUMERATED" => integer = false,
+                                        "INTEGER" => integer = true,
+                                        s => abort!(lit, "`type = \"{}\"` is unsupported", s),
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
```

Comment:
This is... less than ideal, but I have some ideas about how to refactor this along with attribute parsing in general which I can do in a followup PR.